File: src/java/pulse/cdm/patient/actions/SEUseInhaler.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package pulse.cdm.patient.actions;

import pulse.cdm.bind.PatientActions.UseInhalerData;

public class SEUseInhaler extends SEConsciousRespirationCommand
{  
  public SEUseInhaler()
  {
  }
  
  public void copy(SEUseInhaler other)
  {
    if(this==other)
      return;
  }
  
  @Override
  public SEUseInhaler getCopy()
  {
    SEUseInhaler command = new SEUseInhaler();
    command.copy(this);
    
    return command;
  }
  
  @Override
  public void reset()
  {
    super.reset();
  }
  
  @Override
  public boolean isValid()
  {
    return true;
  }
  
  public static void load(UseInhalerData src, SEUseInhaler dst)
  {
    //SEConsciousRespirationCommand.load(src.getConsciousRespirationCommand(), dst);
  }
  
  public static UseInhalerData unload(SEUseInhaler src)
  {
    UseInhalerData.Builder dst = UseInhalerData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  
  protected static void unload(SEUseInhaler src, UseInhalerData.Builder dst)
  {
    //SEConsciousRespirationCommand.unload(src, dst.getConsciousRespirationCommand());
  }
  
  @Override
  public String toString()
  {
    return "Use Inhaler";
  }
}

File: src/cpp/cdm/patient/actions/SEUseInhaler.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "cdm/CommonDefs.h"
#include "cdm/patient/actions/SEUseInhaler.h"
#include "cdm/substance/SESubstance.h"
#include "cdm/properties/SEScalarMass.h"
#include "cdm/properties/SEScalarVolume.h"
#include "cdm/io/protobuf/PBPatientActions.h"

SEUseInhaler::SEUseInhaler(Logger* logger) : SEConsciousRespirationCommand(logger)
{
  
}

SEUseInhaler::~SEUseInhaler()
{

}

void SEUseInhaler::Clear()
{
  SEConsciousRespirationCommand::Clear();
}

void SEUseInhaler::Copy(const SEUseInhaler& src)
{
  PBPatientAction::Copy(src, *this);
}

bool SEUseInhaler::IsValid() const
{
  return SEConsciousRespirationCommand::IsValid();
}

bool SEUseInhaler::IsActive() const
{
  return SEConsciousRespirationCommand::IsActive();
}

void SEUseInhaler::ToString(std::ostream &str) const
{
  str << "Use Inhaler";
  if (HasComment())
    str << "\n\tComment: " << m_Comment;  
  str << std::flush;
}